### DISCUSSION TITLE
Cannot run GUI at all in Ubuntu 18.04

### DISCUSSION THREAD

**Discussion by RickoNoNo3** [pawelsalawa/sqlitestudio#3825] [Q&A]
### Details
Tested in two places: **my WSL** and **an Ubuntu server**
See more in Steps.

#### Step 1
Whether you choose installing or downloading the .tar.xz file, make it done and try to run sqlitestudio with double-click in explorer.
You would see nothing.

#### Step 2
Try to call sqlitestudio by using terminal. Type `./sqlitestudio` into terminal and exec it. The result is:
```
[1]     10000     segmentation fault (core dumped)     ./sqlitestudio
```

#### Step 3
Try CLI mode. Type `./sqlitestudiocli` into terminal and exec it. The result is:
```
./sqlitestudiocli: error while loading shared libraries: libreadline.so.6: cannot open shared object file: No such file or directory
```
#### Step 4
Then we have to **INSTALL `libreadline6`**. (For me it isn't in the apt repository and I used dpkg instead.)

#### Step 5
Retry to run CLI again. It works well now. **BUT**

When exited the CLI process(.quit command), the system still show:
```
[1]     10000     segmentation fault (core dumped)     ./sqlitestudiocli
```

#### Step 6
Retry to run GUI again. The problem still exists.

# I am now on Step 6, and don't know how to do next.

### Operating system
|OS|DE|Arch|
|:--:|:--:|:--:|
|Ubuntu 18.04 (WSL)|Xfce 4|amd64|
|Ubuntu Server 18.04 LTS|Xfce 4|amd64|

problems are same.

### SQLiteStudio version
latest release (3.2.1)

**Comment by RickoNoNo3** [comment]
With the mentality of giving it a try, I downloaded 3.1.0.

**3.1.0 DOESN'T HAVE THIS PROBLEM**

**Comment by pawelsalawa** [comment]
Can you tell me that this command gives you when you execute it against 3.2.1 version?
```
ldd sqlitestudio
```

It would also be nice to see what is says against 3.1.0 - is there a difference?

For me it shows this:
```
~/Downloads/SQLiteStudio$ ldd sqlitestudio 
	linux-vdso.so.1 (0x00007ffd17edc000)
	libcoreSQLiteStudio.so.1 => /home/user/Downloads/SQLiteStudio/./lib/libcoreSQLiteStudio.so.1 (0x00007f3f50002000)
	libguiSQLiteStudio.so.1 => /home/user/Downloads/SQLiteStudio/./lib/libguiSQLiteStudio.so.1 (0x00007f3f4f941000)
	libQt5Widgets.so.5 => /home/user/Downloads/SQLiteStudio/./lib/libQt5Widgets.so.5 (0x00007f3f4f108000)
	libQt5Network.so.5 => /home/user/Downloads/SQLiteStudio/./lib/libQt5Network.so.5 (0x00007f3f4ed72000)
	libQt5Core.so.5 => /home/user/Downloads/SQLiteStudio/./lib/libQt5Core.so.5 (0x00007f3f4e624000)
	libstdc++.so.6 => /usr/lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007f3f4e29b000)
	libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007f3f4e083000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f3f4dc92000)
	libsqlite3.so.0 => /home/user/Downloads/SQLiteStudio/./lib/libsqlite3.so.0 (0x00007f3f4d983000)
	libQt5Script.so.5 => /home/user/Downloads/SQLiteStudio/./lib/libQt5Script.so.5 (0x00007f3f4d4fb000)
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007f3f4d2dc000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f3f4cf3e000)
	libQt5Gui.so.5 => /home/user/Downloads/SQLiteStudio/./lib/libQt5Gui.so.5 (0x00007f3f4c758000)
	libGL.so.1 => /usr/lib/x86_64-linux-gnu/libGL.so.1 (0x00007f3f4c4cc000)
	libz.so.1 => /lib/x86_64-linux-gnu/libz.so.1 (0x00007f3f4c2af000)
	libicui18n.so.56 => /home/user/Downloads/SQLiteStudio/./lib/libicui18n.so.56 (0x00007f3f4be16000)
	libicuuc.so.56 => /home/user/Downloads/SQLiteStudio/./lib/libicuuc.so.56 (0x00007f3f4ba5e000)
	libicudata.so.56 => /home/user/Downloads/SQLiteStudio/./lib/libicudata.so.56 (0x00007f3f4a07b000)
	libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007f3f49e77000)
	libgthread-2.0.so.0 => /usr/lib/x86_64-linux-gnu/libgthread-2.0.so.0 (0x00007f3f49c75000)
	libglib-2.0.so.0 => /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0 (0x00007f3f4995e000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f3f5051e000)
	libGLX.so.0 => /usr/lib/x86_64-linux-gnu/libGLX.so.0 (0x00007f3f4972d000)
	libGLdispatch.so.0 => /usr/lib/x86_64-linux-gnu/libGLdispatch.so.0 (0x00007f3f49477000)
	libpcre.so.3 => /lib/x86_64-linux-gnu/libpcre.so.3 (0x00007f3f49205000)
	libX11.so.6 => /usr/lib/x86_64-linux-gnu/libX11.so.6 (0x00007f3f48ecd000)
	libxcb.so.1 => /usr/lib/x86_64-linux-gnu/libxcb.so.1 (0x00007f3f48ca5000)
	libXau.so.6 => /usr/lib/x86_64-linux-gnu/libXau.so.6 (0x00007f3f48aa1000)
	libXdmcp.so.6 => /usr/lib/x86_64-linux-gnu/libXdmcp.so.6 (0x00007f3f4889b000)
	libbsd.so.0 => /lib/x86_64-linux-gnu/libbsd.so.0 (0x00007f3f48686000)
	librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x00007f3f4847e000)
```

**Comment by pawelsalawa** [comment]
Regarding CLI (`./sqlitestudiocli: error while loading shared libraries: libreadline.so.6: cannot open shared object file: No such file or directory`) you need to have libreadline installed.

If you do, see what `locate libreadline.so` tells you.

**Comment by RickoNoNo3** [comment]
## 3.2.1 VS 3.2.1(yours)
![21vs21](https://user-images.githubusercontent.com/42743884/88627767-6f312e80-d0df-11ea-803f-9f784eef40e8.jpg)

Essentially could not see anything different (except the directory location).

## 3.1.0 VS 3.2.1
![10vs21](https://user-images.githubusercontent.com/42743884/88627772-722c1f00-d0df-11ea-92ca-6c532ba7b1d0.jpg)

may diff a lot

**Comment by pawelsalawa** [comment]
Hm. Looks okay.
How about deleting configuration file to start clear? (I don't know why it would matter, but it's worth to give it a chance).
Config file is kept in `~/.config/sqlitestudio/` and is named `settings3`. You can delete it or rename it temporarily and then try again.
